dip-0003/network-info.md

```diff
@@ -0,0 +1,247 @@
+| platformP2PPort  | uint_16 | 0 or 2 | TCP port of Dash Platform peer-to-peer communication between nodes (network byte order). |
+| platformHTTPPort | uint_16 | 0 or 2 | TCP port of Platform HTTP/API interface (network byte order).                            |
+
+#### <a name="plinfo_rules">Validation Rules</a>
+
+Platform information is invalid if any of these conditions are true:
+
+  1. `platformP2PPort`, `platformHTTPPort` and [masternode information](#masternode-information) fields are not distinct
+  2. `platformP2PPort` and `platformHTTPPort` aren't valid port values [1, 65535]
+  3. For mainnet evonodes, `platformP2PPort` and `platformHTTPPort` aren't set to the default values
+    1. Default Platform P2P port: 26656
+    2. Default Platform HTTP port: 443
+
+## Extended Format
+
+The fields below are represented as `networkInfo` and subsume the responsibility of `platformInfo`.
+Applicability of fields based on masternode type will be defined by the field individually.
+
+| Field     | Type    | Size     | Description                                                                                    |
+| --------- | ------- | -------- | ---------------------------------------------------------------------------------------------- |
+| p_count   | uint_8  | 1        | Number of purposes for which addresses are defined                                             |
+| p_entries | byte[]  | variable | Array of pairings between purpose codes and arrays of addresses containing network information |
+
+### `p_count` field
+
+* The value of this field MUST correspond to the number of [`p_entries`](#p_entries-field)
+
+### `p_entries` field
+
+This field is an array of [`p_count`](#p_count-field) elements of type [`p_entry`](#p_entry-type) where:
+
+* `p_entries[0]`
+    * MUST have [`purpose`](#p_entrypurpose-field) code `CORE_P2P`
+    * MUST be registered regardless of masternode type (as defined in [Appendix B](masternode-types.md)).
+    * [`entries[0]`](#p_entryentries-field) MUST have a [`type`](#entrytype-field) of `0x01` (IPv4 address).
+    * [`entries`](#p_entryentries-field) MUST NOT have a [`type`](#entrytype-field) of `0xD0` (domain name).
+* `p_entries[1]`
+    * MUST have [`purpose`](#p_entrypurpose-field) code `PLATFORM_P2P`.
+    * MUST be registered for EvoNodes (type 1).
+    * MUST NOT be registered for regular masternodes (type 0).
+    * [`entries[0]`](#p_entryentries-field) MUST have a [`type`](#entrytype-field) of `0x01` (IPv4 address).
+    * [`entries`](#p_entryentries-field) MUST NOT have a [`type`](#entrytype-field) of `0xD0` (domain name).
+* `p_entries[2]`
+    * MUST have [`purpose`](#p_entrypurpose-field) code `PLATFORM_API`.
+    * MUST be registered for EvoNodes (type 1).
+    * MUST NOT be registered for regular masternodes (type 0).
+    * [`entries`](#p_entryentries-field) MAY have a [`type`](#entrytype-field) of `0xD0` (domain name).
```

Comment:
It may have IPv4 and IPv6 as well, right?

Reply:
Yes,  `PLATFORM_API` has the option to use `0xD0` (while others don't), they're still eligible to use anything else though we place a restriction that the first entry must be IPv4 or IPv6 and any subsequent entry must resolve to the first entry.